audio/vcvrack/Portfile

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        VCVRack Rack 1.1.6 v
+name                vcvrack
+
+categories          audio
+platforms           darwin
+license             GPL-3
+
+set rack_fundamental_version    1.3.1
```

Comment:
Would it make sense to have a separate vcvrack-fundamental port that vcvrack would depend on? That way it could have its own version number, could be updated independently of vcvrack, could use a normal distfile which could be mirrored, etc.

Reply:
I had that passing thought, but the problem is that the Fundamental plugins have no place on the installed filesystem.

They are only meant as extra source files going into `vcvrack`'s build process.  If I make them a sub-port then they will have to be installed somewhere in the formal filesystem - but once `vcvrack` is built and package, they serve no purpose....

I suppose that's the entire point of `depends_build, I guess 😅

Reply:
> They are only meant as extra source files going into `vcvrack`'s build process

Oh I didn't realize that. I just saw the code in the portfile that just seemed to be fetching the fundamental distfile, verifying checksums, extracting, and then creating a new zip file. I assumed that new zip file was being installed as is. I didn't realize the build process was unpacking it again and building it. I wonder why they make you create a new zip file rather than using the one that was downloaded.

Reply:
Basically the way it works is that Rack can build plugins against its source. There are the set of "fundamental" plugins, and then others/3rd-party.

The fundamental plugins are downloaded as a zip file containing source, extracted into place in the source tree, and after the first phase of Rack's build process, the plugins are then built against the libs and assets of Rack. Afterwards the resulting objects and dylibs (and source code, manifest JSON and what-have-you) are zipped up again and placed into the resulting .app bundle for `vcvrack`.

When the user first double-clicks on the app, Rack creates a user directory in ~/Documents, and extracts the built fundamentals zip into a plugins subdirectory there.

Reply:
Well thanks for figuring this all out. A friend of mine has been raving about the wonders of the Eurorack so I'll have to try this port.

Reply:
No worries. The one thing this is missing is the SDK, which I'll add at some later point in time. The `vcvrack` SDK is necessary to build additional plugins for Rack which also get installed to the user plugins directory.